packages/contracts/src/test/plugin/Cloneable/PluginCloneableSetupMock.sol

```diff
@@ -35,30 +31,40 @@ contract PluginCloneableSetupV1Mock is PluginSetup {
         (_dao, _payload);
         permissions = mockPermissions(5, 6, PermissionLib.Operation.Revoke);
     }
+}
+
+contract PluginCloneableSetupV1MockBad is PluginSetup {
+    constructor(address implementation) PluginSetup(implementation) {}
 
     /// @inheritdoc IPluginSetup
-    function implementation() external view virtual override returns (address) {
-        return address(pluginBase);
+    function prepareInstallation(
+        address _dao,
+        bytes memory
+    ) public virtual override returns (address plugin, PreparedSetupData memory preparedSetupData) {
+        plugin = mockPluginProxy(implementation(), _dao);
+        preparedSetupData.helpers = mockHelpers(1);
+        preparedSetupData.permissions = mockPermissions(5, 6, PermissionLib.Operation.Grant);
     }
-}
 
```

Comment:
Might be missing something but this looks identical to the mock above, what makes it 'bad'?

Reply:
@jordaniza We have to refactor those in task https://aragonassociation.atlassian.net/browse/OS-1229